android/modules/calendar/src/java/ti/modules/titanium/calendar/RecurrenceRuleProxy.java

```diff
@@ -0,0 +1,360 @@
+		calculateDaysOfTheYear();
+		calculateEnd();
+		calculateInterval();
+		calculateMonthsOfTheYear();
+		calculateWeeksOfTheYear();
+	}
+
+	//region Methods for converting native to Kroll values
+	private void calculateDaysOfTheMonth(){
+		String days = matchExpression(".*(BYMONTHDAY=[0-9]*).*", 11);
+		if (days != null && frequency == CalendarModule.RECURRENCEFREQUENCY_MONTHLY) {
+			daysOfTheMonth = new int[1];
+			daysOfTheMonth[0] = Integer.valueOf(days);
+		} else {
+			daysOfTheMonth = new int[]{};
+		}
+	}
+
+	private void calculateDaysOfTheWeek() {
+		// Create a dictionary in the context of a month.
+		if (frequency == CalendarModule.RECURRENCEFREQUENCY_MONTHLY) {
+			String byDay = matchExpression(".*(BYDAY=[,0-9A-Z]*).*", 6);
+			if (byDay != null) {
+				KrollDict daysOfTheWeekDictionary = new KrollDict();
+				daysOfTheWeek = new KrollDict[1];
+				daysOfTheWeekDictionary.put(dayOfWeekKey, weekdaysMap.get(byDay.substring(byDay.length() - 2)));
+				daysOfTheWeekDictionary.put(weekNumberKey, byDay.substring(0, byDay.length() - 2));
+				daysOfTheWeek[0] = daysOfTheWeekDictionary;
+			}
+			return;
+		}
+		// Create a dictionary in the context of a week.
+		if (frequency == CalendarModule.RECURRENCEFREQUENCY_WEEKLY) {
+			String byDay = matchExpression(".*(BYDAY=[,0-9A-Z]*).*", 6);
+			// Split the days from result.
+			String[] days = byDay.split(",");
+			daysOfTheWeek = new KrollDict[days.length];
+			for (int i=0; i < daysOfTheWeek.length; i++) {
+				KrollDict daysOfTheWeekDictionary = new KrollDict();
+				daysOfTheWeekDictionary.put(dayOfWeekKey, weekdaysMap.get(days[i]));
+				// In the context of a weekly recurrence week number is irrelevant.
+				daysOfTheWeekDictionary.put(weekNumberKey, 0);
+				daysOfTheWeek[i]=daysOfTheWeekDictionary;
+			}
+			return;
+		}
+		daysOfTheWeek = new KrollDict[]{};
+	}
+
+	private void calculateDaysOfTheYear() {
+		String days = matchExpression(".*(BYYEARDAY=[0-9]*).*", 10);
+		if (days != null && frequencyMap.get(frequency).equals(CalendarModule.RECURRENCEFREQUENCY_YEARLY)) {
+			daysOfTheYear = new int[1];
+			daysOfTheYear[0] = Integer.valueOf(days);
+		} else {
+			daysOfTheYear = new int[]{};
+		}
+	}
+
+	private void calculateEnd() {
+		// Check for until specific date condition.
+		String date = matchExpression(".*(UNTIL=[0-9A-Z]*).*", 6);
+		SimpleDateFormat sDateFormat = new SimpleDateFormat("yyyyMMdd");
+		if (date != null) {
+			try {
+				endDictionary.put(until, sDateFormat.parse(date.substring(0, 8)));
+			} catch (ParseException e) {
+				e.printStackTrace();
+			}
+		}
+
+		// Check for repeat count.
+		String until = matchExpression(".*(COUNT=[0-9]*).*", 6);
+		if (until != null) {
+			endDictionary.put(count, until);
+		}
+	}
+
+	private void calculateInterval() {
+		String interval = matchExpression(".*(INTERVAL=[0-9]*).*", 9);
+		if (interval != null) {
+			this.interval = Integer.valueOf(interval);
+		}
+	}
+
+	private void calculateMonthsOfTheYear() {
```

Comment:
combine with `calculateWeeksOfTheYear` since they both really apply to yearly frequencies (and maybe rename to something like `calculateYearlyFrequency`)